src/sounds.cpp

```diff
@@ -900,15 +900,16 @@ void sfx::do_ambient()
         play_ambient_variant_sound( "environment", "indoors", heard_volume, channel::indoors_env, 1000 );
     }
 
-    weather_type current_weather = get_weather().weather;
+    weather_datum current_weather = get_weather().data();
```

Comment:
I haven't checked to see if it's modified anywhere here, but I think it'd be a problem if sound is somehow modifying it.
```suggestion
    const weather_datum &current_weather = get_weather().data();
```